core/shared/src/main/scala/zio/config/syntax/ConfigSyntax.scala

```diff
@@ -343,8 +344,7 @@ trait ConfigSyntax {
     //        override def flatten: IndexedFlat = indexedFlat
     //      }
 
-
-    //Disabled until next version of ZIO: https://github.com/zio/zio-config/blob/avoid_custom_index_until_3/README.md#indexed-map-array-datatype-and-a-some-implementation-notes
+    // Disabled until next version of ZIO: https://github.com/zio/zio-config/blob/avoid_custom_index_until_3/README.md#indexed-map-array-datatype-and-a-some-implementation-notes
```

Comment:
Looks like the change was unrelated, but this link appears to be dead.

Reply:
I guess the whole function is obsolete as `ConfigProvider.fromMap` is builtin now.

Reply:
Or maybe it's still relevant. The section seems to be moved to here:
https://github.com/zio/zio-config/blob/v4.0.0-RC15/docs/read-from-various-sources.md#indexed-map-array-datatype-and-a-some-implementation-notes